src/main/java/com/oltpbenchmark/benchmarks/templated/procedures/GenericQuery.java

```diff
@@ -36,9 +36,16 @@ public abstract class GenericQuery extends Procedure {
 
     /** Execution method with parameters. */
     public void run(Connection conn, List<Object> params) throws SQLException {
-        try (PreparedStatement stmt = getStatement(conn, params); ResultSet rs = stmt.executeQuery()) {
-            while (rs.next()) {
-                //do nothing
+        PreparedStatement stmt = getStatement(conn, params);
+        String queryString = stmt.toString().toLowerCase();
+        if (queryString.startsWith("update") || queryString.startsWith("delete") || queryString.startsWith("insert")) {
```

Comment:
Is the problem that this type of statement doesn't produce a ResultSet? If so, a try/catch solution might be easier than filtering for specific strings. I.e., try executeQuery first, if an exception is thrown, try execute (I think that would return just a boolean true/false whether the query executed successfully). If that's true, yey!, if not, throw an error?